sprint_mission_9/backend/tests/integration/products.test.ts

```diff
@@ -0,0 +1,439 @@
+
+      expect(response.body.data.length).toBeLessThanOrEqual(2);
+      expect(response.body).toHaveProperty('total');
+      expect(response.body).toHaveProperty('page');
+      expect(response.body).toHaveProperty('totalPages');
+    });
+
+    it('should include user info and counts', async () => {
+      const response = await request(app).get('/api/products').expect(200);
+
+      if (response.body.data.length > 0) {
+        const product = response.body.data[0];
+        expect(product).toHaveProperty('user');
+        expect(product.user).toHaveProperty('nickname');
+        expect(product).toHaveProperty('likeCount');
+        expect(product).toHaveProperty('commentCount');
+      }
+    });
+  });
+
+  describe('인증 불필요 - GET /api/products/:id', () => {
+    it('should return a product by id', async () => {
+      const response = await request(app).get('/api/products/1').expect(200);
+
+      expect(response.body).toHaveProperty('id');
+      expect(response.body).toHaveProperty('name');
+      expect(response.body).toHaveProperty('price');
+    });
+
+    it('should return 404 for non-existent product', async () => {
+      const response = await request(app).get('/api/products/999999').expect(404);
+
+      expect(response.body).toHaveProperty('message');
+      expect(response.body.message).toBe('Product not found');
+    });
+
+    it('should include user info and stats', async () => {
+      const response = await request(app).get('/api/products/1').expect(200);
+
+      expect(response.body).toHaveProperty('user');
+      expect(response.body.user).toHaveProperty('nickname');
+      expect(response.body).toHaveProperty('likeCount');
+      expect(response.body).toHaveProperty('commentCount');
+    });
+  });
+
+  describe('인증 필요 - POST /api/products', () => {
+    let authToken: string;
+    let testUserId: number;
+
+    beforeAll(async () => {
+      const user = await prisma.user.create({
+        data: {
+          email: `product-create-test-${Date.now()}@example.com`,
+          password: 'hashedpassword',
+          nickname: 'Product Create Test User',
+        },
+      });
+      testUserId = user.id;
+      authToken = generateAuthToken(testUserId);
+    });
+
+    afterAll(async () => {
+      await prisma.product.deleteMany({ where: { userId: testUserId } });
+      await prisma.user.deleteMany({ where: { id: testUserId } });
+    });
+
+    it('should create a product with authentication', async () => {
+      const newProduct = {
+        name: 'Test Product',
+        description: 'Test Description',
+        price: 10000,
+        tags: ['test'],
+      };
+
+      const response = await request(app)
+        .post('/api/products')
+        .set('Authorization', `Bearer ${authToken}`)
+        .send(newProduct);
+
+      if (response.status === 201) {
+        expect(response.body).toHaveProperty('id');
+        expect(response.body.name).toBe(newProduct.name);
+      }
```

Comment:
이렇게 테스트를 작성하면 API 실패시에 알 수 없어요. supertest에서 명시적으로 201을 던지고, 다른 검증은 상태값을 통한 분기를 하지 않고 검증 진행해주세요.


```suggestion
      const response = await request(app)
        .post('/api/products')
        .set('Authorization', `Bearer ${authToken}`)
        .send(newProduct)
        .expect(201); // 명시적으로 상태코드 검증

      expect(response.body).toHaveProperty('id');
      expect(response.body.name).toBe(newProduct.name);
```